### ISSUE TITLE
Add a Definition of Done Section to the Issue Templates

### ISSUE THREAD

**Issue by jlaura** [DOI-USGS/ISIS3#3423]
**Description**

The issue templates for bugs, docs, and issues should have a definition of done section soliciting input from the issue poster on when they consider the issue to be closed. Right now, this is implicitly being assumed and we are seeing misalignment in expectations.

**Suggested Edit**

Add a DoD section to the three necessary templates.

Thanks!

**Comment by jessemapel** [comment]
We can clarify that the DoD for bugs is that the reproduction steps no longer produce the bug. Then, on features we need a definition of done.

**Comment by jlaura** [comment]
Yup. I was meaning to post this today, but did not make it to it. We already have a DoD on bugs. 

When a developer is no longer able to reproduce using the provided steps, CI passes, and the PR is merged the bug should be closed. If anyone is able to reproduce using the provided steps, we should get the bug back open (this could be the reporter, another committer, another contributor, a new contributor, etc.). If the bug causes a follow-on problem, we need to get a new bug report up with new steps to reproduce.

Same with documentation issues. On those, we need to make sure that before we start a documentation issue, we have an explicit description of how specifically the documentation needs to be improved. A quick skim of the current issues tagged docs suggests that this is a non-issue and that the template is working fine for now. 

For a feature enhancement, I would almost suggest modifying the template more than just DoD. We all benefit from a good user story, so we should prompt users in the template to provide a user story in the form `As a <role/person>, I can <goal> so that <why>` or `As a <some class of user>, I want to <be able to do/achieve something> so that <value added thing happens>`.  We can get this into the 'Description' section of the template. 

I am honestly not sure the best way to solicit a good DoD in a single shot because a good DoD is hard. Additionally, we do not have a single product owner interfacing with the team who is going to give the thumbs-up/thumbs-down on the user story. We do not need to include any items in here on the dev. process that we see so frequently in DoDs. We know that things like CI, review, tests, etc. are going to be completed. (In the case of tests and Q-applications we do not have a good solution at this time.)

We could go with two acceptance criteria sections and solicit functional and non-functional acceptance criteria. Functional are standard, the feature shall do XYZ. For non-functional I am thinking the qualitative items: (1) for XYZ data, (2) 99% of the time, (3) with performance under some quantitative threshold, (4) usable by someone of XYZ experience w.r.t. the complexity of the tasks being accomplished, etc. So the template could prompt with suggestions on what functional/non-functional criteria would look like for a given feature.

The examples in the template also need to be expressive enough that we describe functional/non-functional requirements for both lower level API enhancements and non-developer user facing enhancements.

**Comment by ascbot** [comment]
I am a bot that cleans up old issues that do not have activity.

Happy Birthday to this issue! :birthday:

Unfortunately, this issue has not received much attention in the last 12 months. Therefore, I am going to close it. Please feel free to reopen this issue or open a new issue sometime in the future. If this issue is a bug report, please check that the issue still exists in our newest version before reopening.
